Add secondary runtime handler and cgroup v2 config to CRIO tests

harche opened the pull request:
This PR adds following things,

1. A secondary runtime handler called, `preconfigured-handler`. As a part of moving the [RuntimeClass to GA](https://docs.google.com/document/d/17nROj6ayPsUpx09mhLrzOkRLgo-8sn6Vgfyy-gmyuo4/edit#heading=h.bro64m4ptb7g), we want to add a second runtime handler that will actually invoke a different runtime to run the tests like [these](https://github.com/kubernetes/kubernetes/blob/b7d44329f3514a65af9048224329a4897cf4d31d/test/e2e/common/runtimeclass.go#L55) instead of just reusing the `runc` under different handler name. In this case, we are going to use [crun](https://github.com/containers/crun) as our secondary runtime. 
2. Add the tests for cgroup v2 functionality - In case of cgroup v2 testing, we use `crun` as our primary runtime while the secondary runtime  `preconfigured-handler` points to `runc`. 

Results from the local testing of the Fedora CoreOS ignition configurations,

Let's test cgroup v1 ignition first,
```bash
[harshal@localhost crio-fcos-config]$ export CONFIG=/home/harshal/go/src/github.com/kubernetes/test-infra/jobs/e2e_node/crio/crio.ign 
[harshal@localhost crio-fcos-config]$ gcloud compute instances create --metadata-from-file "user-data=${CONFIG}" --image-project "fedora-coreos-cloud" --image-family "fedora-coreos-${STREAM}" "${VM_NAME}-1"
Created [https://www.googleapis.com/compute/v1/projects/openshift-gce-devel/zones/us-east1-c/instances/harpatil-ci-1].
NAME           ZONE        MACHINE_TYPE   PREEMPTIBLE  INTERNAL_IP  EXTERNAL_IP     STATUS
harpatil-ci-1  us-east1-c  n1-standard-1               10.240.0.94  35.196.241.173  RUNNING

[root@harpatil-ci-1 core]# stat -f -c %T /sys/fs/cgroup/
tmpfs
[root@harpatil-ci-1 core]# systemctl status crio | grep Active
     Active: active (running) since Thu 2020-09-17 05:00:22 UTC; 1min 54s ago
[root@harpatil-ci-1 core]# 
```
Corresponding runtime configuration in CRIO looks like,
```toml
[crio.runtime.runtimes.preconfigured-handler]
runtime_path = "/usr/local/bin/crun"
runtime_type = ""
runtime_root = ""

[crio.runtime.runtimes.runc]
runtime_path = ""
runtime_type = "oci"
runtime_root = "/run/runc"
``` 
with `default_runtime = "runc"`

So far so good, now let's test cgroup v2 ignition, 
```bash
export CONFIG=/home/harshal/go/src/github.com/kubernetes/test-infra/jobs/e2e_node/crio/crio_cgrpv2.ign 
[harshal@localhost crio-fcos-config]$ gcloud compute instances create --metadata-from-file "user-data=${CONFIG}" --image-project "fedora-coreos-cloud" --image-family "fedora-coreos-${STREAM}" "${VM_NAME}-1"
Created [https://www.googleapis.com/compute/v1/projects/openshift-gce-devel/zones/us-east1-c/instances/harpatil-ci-1].
NAME           ZONE        MACHINE_TYPE   PREEMPTIBLE  INTERNAL_IP  EXTERNAL_IP     STATUS
harpatil-ci-1  us-east1-c  n1-standard-1               10.240.0.53  35.237.226.136  RUNNING

[root@harpatil-ci-1 core]# stat -f -c %T /sys/fs/cgroup/
cgroup2fs
[root@harpatil-ci-1 core]# systemctl status crio  |grep Active
     Active: active (running) since Thu 2020-09-17 05:07:12 UTC; 1min 27s ago
```
Corresponding runtime configuration in CRIO looks like,
```toml
[crio.runtime.runtimes.crun]
runtime_path = "/usr/local/bin/crun"
runtime_type = ""
runtime_root = ""

[crio.runtime.runtimes.preconfigured-handler]
runtime_path = "/usr/local/bin/runc"
runtime_type = ""
runtime_root = ""
```
with `default_runtime = "crun"` 

In case you want to review the machine friendly FCOS ignition files closely, human friendly corresponding YAMLs are kept [here](https://github.com/harche/crio-fcos-config) with the filenames ending with `*.fcc` 

One final note, the test jobs in this PR won't run successfully unless https://github.com/kubernetes/kubernetes/pull/94760 is merged. Because we use Fefora CoreOS, we need to make sure the username used to access the host is set to the value `core`. 

 
Signed-off-by: Harshal Patil <harpatil@redhat.com>

harche commented:
cc @sjenning @mrunalp @giuseppe @SergeyKanzhelev @dims

harche commented:
/sig node

harche commented:
Once the https://github.com/kubernetes/kubernetes/pull/94760 is merged, and new kubekins image is built, I will update his PR to point to the new kubekins image with that fix.

harche commented:
/assign @dashpole

SergeyKanzhelev commented:
/unassign @dashpole

@dashpole is switching his focus.

@harche is it still WIP?

harche commented:
@SergeyKanzhelev yes, because I am waiting for kubekins build to pick up my changes.

https://github.com/kubernetes/test-infra/pulls?q=is%3Apr+is%3Amerged+kubekins+

Once it's my changes make it in kubekins image, I will update the test jobs in this PR to use that new image.

harche commented:
/assign @harche

harche commented:
@SergeyKanzhelev @dims I have removed the `WIP` from this PR. I have updated the kubekins images used by the jobs to `gcr.io/k8s-testimages/kubekins-e2e:v20200918-58bdf8d-master` which seems to be having my changes, https://github.com/kubernetes/test-infra/pull/19288

dims commented:
/ok-to-test
/assign @sjenning 
/approve

dims commented:
/lgtm
